### PR TITLE
[markFeatureWriter] skip ignorable anchors

### PR DESCRIPTION
**Pull request by moyogo**
Glyphs.app doesn't export anchors with names starting with "#" or "_#".

See also https://github.com/googlefonts/ufo2ft/issues/477

### CODE REVIEW

**Comment by khaledhosny** [comment]
Shouldn’t this be handled somehow by glyphsLib? I don’t think ufo2ft imposing arbitrary Glyphs conventions on any UFO file. May be ufo2ft should have some this-is-a-UFO-converted-from-Glyphs-file flag to apply this and other Glyph-specific conventions.

**Comment by moyogo** [comment]
> Shouldn’t this be handled somehow by glyphsLib? I don’t think ufo2ft imposing arbitrary Glyphs conventions on any UFO file. 

We can either add an anchor objectLib key to not export a specific anchor or a font.lib key to not export anchors matching a no-export list.

> May be ufo2ft should have some this-is-a-UFO-converted-from-Glyphs-file flag to apply this and other Glyphs-specific conventions.

That could work a well. 

I'm not sure what to do.
@anthrotype @madig any thoughts?

**Comment by anthrotype** [comment]
it may be arbitrary and Glyphs.app-only, but it's convenient, even in a UFO only workflow. ObjectLibs could work as well, but they are clumsy to use (is there a UI for them?).
I'm leaning towards having this in.
How common do you think people name their anchors starting with `#` who do _not_ want them to be ignorable and may be disrupted by this change? Very few, if any, I reckon.

**Comment by anthrotype** [comment]
@moyogo perhaps we can add an `ignorePrefix` option to the MarkFeatureWriter and set that to `#` (via the UFO.lib featureWriters key) when glyphsLib exports UFOs from .glyphs sources? So we won't disrupt UFO-only workflow that may happen to be using `#` to _not_ mean "skip this anchor from GPOS"

**Comment by khaledhosny** [comment]
I think this is better done in glyphsLib minimal mode now, and fontmake should start using it. This way, Glyphs-specific behavior is localized in glyphsLib.

**Comment by madig** [comment]
Skipping "#" is not enough, https://glyphsapp.com/learn/reusing-shapes-component-tricks#g-__repurposing-cursive-attachment-for-ligatures says you can use any non-letter character. 

> I think this is better done in glyphsLib minimal mode now, and fontmake should start using it. This way, Glyphs-specific behavior is localized in glyphsLib.

Mh, yeah, but then you still need something for compiling UFOs that contain these anchors, because you might keep them in your sources and compile from there. Maybe an opt-in filter would do the job.

**Comment by anthrotype** [comment]
+1 for an opt-in filter, always turned on for glyphsLib-generated UFOs

**Comment by khaledhosny** [comment]
glyphsLib now has its own mark feature writer, so this should be done there now.

File: Lib/ufo2ft/featureWriters/markFeatureWriter.py [hunk 1]
```diff
@@ -326,6 +326,10 @@ def _getAnchorLists(self):
                         "unnamed anchor discarded in glyph '%s'", glyphName
                     )
                     continue
+                # Skip anchors not meant for export
+                # See https://github.com/googlefonts/ufo2ft/issues/477
+                if anchorName.startswith(("#", "_#")):
```
Comment (khaledhosny) [inline Lib/ufo2ft/featureWriters/markFeatureWriter.py:331]: This should check for anchors started by any non-letter character, not only `#`.

**Comment by khaledhosny** [comment]
https://github.com/googlefonts/glyphsLib/pull/964

This PR should either be closed or changed to fix anchor names that are invalid for feature files (instead of skipping them silently, which is a Glyphs thing) or even just error.

**Comment by khaledhosny** [comment]
Fixed in #869

**Pull request closed without merging**